### PR TITLE
getIntersectionPoint failed when there is no intersection on the terrain

### PR DESCRIPTION
**Pull request by super626**

### CODE REVIEW

File: cocos/3d/CCTerrain.cpp [hunk 1]
```diff
@@ -327,6 +327,10 @@ float Terrain::getHeight(float x, float z, Vec3 * normal)
     if(image_x>=_imageWidth-1 || image_y >=_imageHeight-1 || image_x<0 || image_y<0)
     {
         return 0;
+        if (normal)
```
Comment (zilongshanren) [inline cocos/3d/CCTerrain.cpp:329]: This line will never be executed... You put a clause after `return 0`

File: cocos/3d/CCTerrain.h [hunk 2]
```diff
@@ -343,6 +343,8 @@ class CC_DLL Terrain :public Node
      */
     Vec3 getIntersectionPoint(const Ray & ray);
 
+    bool getIntersectionPoint(const Ray & ray, Vec3 & intersectionPoint);
```
Comment (zilongshanren) [inline cocos/3d/CCTerrain.h]: please add `const` at the end of `getIntersectonPoint`

File: cocos/3d/CCTerrain.h [hunk 3]
```diff
@@ -341,7 +341,9 @@ class CC_DLL Terrain :public Node
      * Ray-Terrain intersection.
      * @return the intersection point
      */
-    Vec3 getIntersectionPoint(const Ray & ray);
+    Vec3 getIntersectionPoint(const Ray & ray) const;
+
+    bool getIntersectionPoint(const Ray & ray, Vec3 & intersectionPoint) const;
```
Comment (zilongshanren) [inline cocos/3d/CCTerrain.h:352]: @super626  please add comment to this method

**Comment by zilongshanren** [comment]
@super626  This PR does two thing. 
1. Fix getIntersectionPoint failed when there is no intersection on the terrain
2. add two APIs for Physic3DWorld?

I think it's better to do one thing in a PR.  You don't need to change this PR. 
But please pay attention to the future's PR.
Thanks.

**Comment by super626** [comment]
@andyque 
Ok. Thanks

**Pull request merged**